addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CrossDomainScriptInclusionScanRule.java

```diff
@@ -51,16 +54,22 @@ public class CrossDomainScriptInclusionScanRule extends PluginPassiveScanner {
             LogManager.getLogger(CrossDomainScriptInclusionScanRule.class);
     private Model model = null;
 
+    private final TrustedDomains trustedDomains = new TrustedDomains();
+
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        trustedDomains.update(getConfig().getString(RuleConfigParam.RULE_DOMAINS_TRUSTED, ""));
+        trustedDomains.add(new RegexTrust(msg.getRequestHeader().getHostName()));
```

Comment:
This can be removed, we are only checking the trusted domains if it's from other domain.

Reply:
I think update is required to refresh the data?

Reply:
I mean line 62 specifically.